cypress/integration/common.js

```diff
@@ -110,6 +110,202 @@ describe('server smoke tests', function() {
+
+        cy.checkA11y();
+
+        cy.seedUser().then(newUser => {
+            logIn(newUser.username, 'invalidpassword');
+            assertError(messageText);
+        });
+    });
+
+    it('should rate-limit users attempting to login too often', () => {
+        const fakeEmail = `${Date.now()}@example.com`;
+        const fakePassword = 'hunter2';
+        const maxAttempts = 10;
+
+        times(maxAttempts, function() {
+            cy.loginUser({
+                username: fakeEmail,
+                password: fakePassword
+            }).then(response => {
+                expect(response.status).to.eq(200);
+            });
+        });
+
+        cy.loginUser({
+            username: fakeEmail,
+            password: fakePassword,
+            failOnStatusCode: false
+        }).then(response => {
+            expect(response.status).to.eq(429);
+            expect(response.body).to.include('Too many requests');
+        });
+    });
+
+    it('should prevent registrations with invalid passwords', () => {
+        const username = `${Date.now()}@example.com`;
+
+        cy.visit('/user/register');
+
+        createAccount(username, '5555555555');
+        cy.getByTestId('form-errors').should('contain', 'Password is too weak');
+
+        // Non-UI tests for remaining validations for speed
+        cy.registerUser({
+            username: username,
+            password: username
+        }).then(res => {
+            expect(res.body).to.contain(
+                'Password must be different from your email address'
+            );
+        });
+
+        cy.registerUser({
+            username: username,
+            password: 'tooshort'
+        }).then(res => {
+            expect(res.body).to.contain('Password must be at least');
+        });
+    });
+
+    it('should register and see activation screen', function() {
+        // Register
+        cy.visit('/user/register');
+        createAccount(`${Date.now()}@example.com`, uuid());
+        cy.checkA11y();
+        cy.get('body').should(
+            'contain',
+            'Check your emails to activate your account'
+        );
+    });
+
+    it('should email valid users with a token', () => {
+        const now = Date.now();
+        const username = `${now}@example.com`;
+        cy.registerUser({
+            username: username,
+            password: `password${now}`,
+            returnToken: true
+        }).then(res => {
+            // via https://github.com/auth0/node-jsonwebtoken/issues/162
+            expect(res.body.token).to.match(
+                /^[a-zA-Z0-9\-_]+?\.[a-zA-Z0-9\-_]+?\.([a-zA-Z0-9\-_]+)?$/
+            );
+            expect(res.body.mailParams.sendTo).to.equal(username);
+            expect(res.body.mailParams.subject).to.equal(
+                'Activate your The National Lottery Community Fund website account'
+            );
+        });
+    });
+
+    it('should be able to log in and update account details', () => {
```

Comment:
This is the most notable addition here, making sure there are e2e tests for updating account details when logged in. The rest of the changes are streamlining the user tests.